src/main/java/com/employeesService/EmployeesService/service/DepartmentServiceImpl.java

```diff
@@ -31,16 +28,18 @@ public Department save(Department department) {
     }
 
     @Override
-    public ResponseEntity<Department> getById(String deptNo) {
+    public Department getById(String deptNo) throws DepartmentNotFoundException {
         try {
-            return new ResponseEntity<>(departmentRepository.findById(deptNo).orElseThrow(), HttpStatus.OK);
+            Department department = departmentRepository.findById(deptNo).orElseThrow();
+            return department;
         } catch (DepartmentNotFoundException d) {
-            throw new ResponseStatusException(HttpStatus.NOT_FOUND, "Department not found with id: " + deptNo);
+            throw new DepartmentNotFoundException("Department not found for Id: " + deptNo);
```

Comment:
The exception should be thrown in a more functional programming way, chaining in the Optional, using its `orElseThrow()` but with your exception!

Did you test this? How did you test this?